test/jndi_mysql_test.rb

```diff
@@ -24,26 +24,22 @@ def setup
 
     def teardown; ActiveRecord::Base.connection_pool.disconnect! end
 
-    test "configures once" do
+    test "configures connection" do
       pool = ActiveRecord::Base.connection_pool
-      assert_false pool.active_connection? if pool.respond_to?(:active_connection?)
+      assert_nil pool.active_connection? if pool.respond_to?(:active_connection?)
 
-      adapter_class.any_instance.expects(:configure_connection).once
+      # called twice since there's a version check during initialize that needs a connection
+      adapter_class.any_instance.expects(:configure_connection).at_least_once
       ActiveRecord::Base.connection.exec_query "SELECT VERSION()"
     end
 
-    test "configures on demand (since it's lazy)" do
```

Comment:
this won't pass? 
not sure how much it matters but the problem with 'older' AR was that a conn was being checkout from the middleware regardless of whether a conn was about to be used during a request ... thus with AR callbacks we worked around such situations by only checking out on first actual use.

Reply:
nope, won't pass...problem is, somewhere in the 5.0 line Rails added a version check in initialize() of the abstract mysql adapter. This version check does nothing more than connect to the db, select the version and complain that it is too old...oh well...for fun I just removed it the Rails code, to have this passing...

dunno...one thing we could do is to set version to a dummy value high enough before calling super, reset it to nil afterwards so the next real call that needs a version does actually load it. I tried that too and it seems to work, but it's a bit dirty. Just noticed that in Rails 6 they moved it to a `check_version` function that can be overriden with nothing...exactly to have it lazy again..

so...I like neither of the options too much, but being really lazy would be good.

Reply:
passes now